sw/airborne/modules/computer_vision/opticflow_module.c

```diff
@@ -109,27 +129,30 @@ void opticflow_module_run(void)
 {
   pthread_mutex_lock(&opticflow_mutex);
   // Update the stabilization loops on the current calculation
-  if (opticflow_got_result) {
-    uint32_t now_ts = get_sys_time_usec();
-    AbiSendMsgOPTICAL_FLOW(FLOW_OPTICFLOW_ID, now_ts,
-                           opticflow_result.flow_x,
-                           opticflow_result.flow_y,
-                           opticflow_result.flow_der_x,
-                           opticflow_result.flow_der_y,
-                           opticflow_result.noise_measurement,
-                           opticflow_result.div_size);
-    //TODO Find an appropriate quality measure for the noise model in the state filter, for now it is tracked_cnt
-    if (opticflow_result.noise_measurement < 0.8) {
-      AbiSendMsgVELOCITY_ESTIMATE(VEL_OPTICFLOW_ID, now_ts,
-                                  opticflow_result.vel_body.x,
-                                  opticflow_result.vel_body.y,
-                                  0.0f, //opticflow_result.vel_body.z,
-                                  opticflow_result.noise_measurement,
-                                  opticflow_result.noise_measurement,
-                                  -1.0f //opticflow_result.noise_measurement // negative value disables filter updates with OF-based vertical velocity.
-                                 );
+  for (int idx_camera = 0; idx_camera < ACTIVE_CAMERAS; idx_camera++) {
+    if (opticflow_got_result[idx_camera]) {
+      uint32_t now_ts = get_sys_time_usec();
+      AbiSendMsgOPTICAL_FLOW(FLOW_OPTICFLOW_ID + idx_camera, now_ts,
+                             opticflow_result[idx_camera].flow_x,
+                             opticflow_result[idx_camera].flow_y,
+                             opticflow_result[idx_camera].flow_der_x,
+                             opticflow_result[idx_camera].flow_der_y,
+                             opticflow_result[idx_camera].noise_measurement,
+                             opticflow_result[idx_camera].div_size,
+                             opticflow_result[idx_camera].camera_id);
+      //TODO Find an appropriate quality measure for the noise model in the state filter, for now it is tracked_cnt
+      if (opticflow_result[idx_camera].noise_measurement < 0.8) {
+        AbiSendMsgVELOCITY_ESTIMATE(VEL_OPTICFLOW_ID, now_ts,
```

Comment:
Can I change my mind on adding the camera id to the VELOCITY_ESTIMATE ABI message? :$
In hindsight, I think it might also be valuable for ABI subscribers to filter out velocity estimates from cameras that might be poorly placed for velocity measurements (e.g. front-facing cameras).

```suggestion
        AbiSendMsgVELOCITY_ESTIMATE(VEL_OPTICFLOW_ID + idx_camera, now_ts,
```
plus reserving ids in `abi_sender_ids.h`, similarly to what you have done for the OPTICAL_FLOW message ;)